tests/std/tests/LWG3987_including_flat_foo_doesnt_provide_begin_end/test_optional.cpp

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <version>
+#if defined(__cpp_lib_optional_range_support) && _HAS_CXX17

```

Comment:
I believe `__cpp_lib_optional_range_support` won't be defined in pre-C++26 modes. Perhaps we should also drop this file at this moment.